crates/api/src/error.rs

```diff
@@ -0,0 +1,127 @@
+//! Kitsune2 error types.
+
+use std::sync::Arc;
+
+/// A clonable trait-object inner error.
+#[derive(Clone, Default)]
+pub struct DynInnerError(
```

Comment:
I'd like the inner errors to be reasonably expressive. Is that what this is allowing? 

Like I could define a `thiserror::Error` in the gossip module and use that internally to define expressive errors for testing but at the API boundary I should use `DynInnerError` so that other modules can use the API without ending up nesting the module errors?

Reply:
Yes, this is exactly the use-case I was hoping for with this.

Reply:
Super, I like it :)

Reply:
@ThetaSinner - Since this comment is specifically on the DynInnerError type, I feel the need to point out: My comment above is actually about the "Other" variant. This DynInnerError type is specifically because the thiserror Derive logic is not expressive enough, I had to have a newtype here or require a custom Display implementation on the K2Error type which would have meant later variants couldn't make use of the handy macros for configuring the variant display impl.

Reply:
Hmm okay, I think I follow but I also think I need to get hands on with this and see how it applies. I'm happy with this as a way to keep errors simple initially and if we end up wanting to make adjustments to the inner details then that is fine :)